iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractConnectable.java

```diff
@@ -206,59 +205,13 @@ public boolean connect(Predicate<Switch> isTypeSwitchToOperate) {
 
     @Override
     public boolean connect(Predicate<Switch> isTypeSwitchToOperate, ThreeSides side) {
-        // ReportNode
-        ReportNode reportNode = this.getNetwork().getReportNodeContext().getReportNode();
-
-        // Booleans
-        boolean isAlreadyConnected = true;
-        boolean isNowConnected = true;
-
-        // Initialisation of a list to open in case some terminals are in node-breaker view
-        Set<SwitchImpl> switchForDisconnection = new HashSet<>();
-
-        // We try to connect each terminal
-        for (TerminalExt terminal : getTerminals(side)) {
-            // Check if the terminal is already connected
-            if (terminal.isConnected()) {
-                reportNode.newReportNode()
-                    .withMessageTemplate("alreadyConnectedTerminal", "A terminal of connectable ${connectable} is already connected.")
-                    .withUntypedValue("connectable", this.getId())
-                    .withSeverity(TypedValue.WARN_SEVERITY)
-                    .add();
-                continue;
-            } else {
-                isAlreadyConnected = false;
-            }
-
-            // If it's a node-breaker terminal, the switches to connect are added to a set
-            if (terminal.getVoltageLevel() instanceof NodeBreakerVoltageLevel nodeBreakerVoltageLevel) {
-                isNowConnected = nodeBreakerVoltageLevel.getConnectingSwitches(terminal, isTypeSwitchToOperate, switchForDisconnection);
-            }
-            // If it's a bus-breaker terminal, there is nothing to do
-
-            // Exit if the terminal cannot be connected
-            if (!isNowConnected) {
-                return false;
-            }
-        }
-
-        // Exit if the connectable is already fully connected
-        if (isAlreadyConnected) {
-            return false;
-        }
-
-        // Connect all bus-breaker terminals
-        for (TerminalExt terminal : getTerminals(side)) {
-            if (!terminal.isConnected()
-                && terminal.getVoltageLevel().getTopologyKind() == BUS_BREAKER) {
-                // At this point, isNowConnected should always stay true but let's be careful
-                isNowConnected = isNowConnected && terminal.connect(isTypeSwitchToOperate);
-            }
-        }
+        List<Terminal> terminalsToConnect = new ArrayList<>(getTerminals(side));
```

Comment:
I don't think you need to copy the list, you only iterate over it in the connect. You did it for the case of the connectable being removed while connecting it?